lib/elixir/lib/kernel.ex

```diff
@@ -2676,10 +2676,10 @@ defmodule Kernel do
     end
 
     case right do
-      _ when in_module? ->
```

Comment:
I think the order is important here as we could be interested in possible side-effects in `unquote(left)`.